cloudflare/data_cloudflare_devices.go

```diff
@@ -0,0 +1,73 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareDevices() *schema.Resource {
+	return &schema.Resource{
+		Schema: resoureceCloudflareDevicesSchema(),
+		Read:   dataResourceCloudflareDevicesRead,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareDevicesImport,
+		},
+	}
+}
+
+func dataResourceCloudflareDevicesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get("account_id").(string)
+
+	devices, err := client.ListTeamsDevices(context.Background(), accountID)
+
+	if err != nil {
+		return fmt.Errorf("error finding devices in account %q: %s", accountID, err)
+	}
+
+	deviceDetails := make([]interface{}, 0)
+
+	for _, device := range devices {
+		deviceDetails = append(deviceDetails, map[string]interface{}{
+			"id":          device.ID,
+			"key":         device.Key,
+			"device_type": device.DeviceType,
+			"name":        device.Name,
+			"version":     device.Version,
+			"updated":     device.Updated,
+			"created":     device.Created,
+			"last_seen":   device.LastSeen,
+			"model":       device.Model,
+			"os_version":  device.OSVersion,
+			"ip":          device.IP,
+			"user_id":     device.User.ID,
+			"user_email":  device.User.Email,
+			"user_name":   device.User.Name,
+		})
+	}
+
+	err = d.Set("devices", deviceDetails)
+	if err != nil {
+		return fmt.Errorf("error setting device details: %s", err)
+	}
+
+	return nil
+}
+
+func resourceCloudflareDevicesImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
you shouldn't need this for a data source as there is no "importing" that happens.